types/index.d.ts

```diff
@@ -19,6 +19,13 @@ declare module 'fastify' {
 type FastifyStaticPlugin = FastifyPluginAsync<NonNullable<fastifyStatic.FastifyStaticOptions>>;
 
 declare namespace fastifyStatic {
+  export interface SetHeadersResponse {
+    getHeader: FastifyReply['getHeader'];
+    setHeader: FastifyReply['header'];
+    readonly filename: string;
+    statusCode: number;
+  }
```

Comment:
This value is defined here:

https://github.com/fastify/fastify-static/blob/5de6d94bf725f8ed8c35b8fbc978335ac02c1f42/index.js#L232-L259

I was a little unclear about the intent and made my best guess.